File: src/01-easy/unary/solution.scala
object Solution extends App {
  val msg = scala.io.StdIn.readLine()
  val bin = msg.map(c => String.format("%7s", Integer.toBinaryString(c.toInt)).replace(' ', '0')).mkString
  val parts = collection.mutable.ListBuffer[String]()
  var i = 0
  while (i < bin.length) {
    val cur = bin(i)
    var count = 0
    while (i < bin.length && bin(i) == cur) { count += 1; i += 1 }
    parts += s"${if (cur == '1') "0" else "00"} ${"0" * count}"
  }
  println(parts.mkString(" "))
}

File: src/01-easy/unary/solution.java
import java.util.*;
class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String msg = sc.nextLine();
        StringBuilder bin = new StringBuilder();
        for (char c : msg.toCharArray())
            bin.append(String.format("%7s", Integer.toBinaryString(c)).replace(' ', '0'));
        String s = bin.toString();
        StringBuilder out = new StringBuilder();
        int i = 0;
        while (i < s.length()) {
            char cur = s.charAt(i);
            int count = 0;
            while (i < s.length() && s.charAt(i) == cur) { count++; i++; }
            if (out.length() > 0) out.append(' ');
            out.append(cur == '1' ? "0" : "00").append(' ').append("0".repeat(count));
        }
        System.out.println(out);
    }
}
